homeassistant/components/dreo/__init__.py

```diff
@@ -0,0 +1,58 @@
+"""Dreo for Integration."""
+import logging
+from dataclasses import dataclass
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DOMAIN, PLATFORMS
+from hscloud.hscloud import HsCloud
+from hscloud.hscloudexception import HsCloudException, HsCloudBusinessException
+
+_LOGGER = logging.getLogger(__name__)
+
+type MyConfigEntry = ConfigEntry[MyData]
+
+
+@dataclass
+class MyData:
+    client: HsCloud
+    devices: []
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: MyConfigEntry):
+    """Set up Dreo from as config entry."""
+    username = config_entry.data[CONF_USERNAME]
+    password = config_entry.data[CONF_PASSWORD]
+
+    manager = HsCloud(username, password)
+    try:
+        await hass.async_add_executor_job(manager.login)
+
+    except HsCloudException as ex:
+        _LOGGER.exception("unable to connect")
+        raise ConfigEntryNotReady(f"unable to connect") from ex
+
+    except HsCloudBusinessException as ex:
+        _LOGGER.exception("invalid username or password")
+        raise ConfigEntryNotReady(f"invalid username or password") from ex
+
+    except Exception as ex:  # pylint: disable=broad-except
+        _LOGGER.exception("Unexpected exception")
+        raise ConfigEntryNotReady(f"Unexpected exception") from ex
+
+    config_entry.runtime_data = MyData(manager, await hass.async_add_executor_job(manager.get_devices))
```

Comment:
Looking at https://github.com/dreo-team/hscloud/blob/300d4983fbd42092b80c8e6a1881d68c64665dbf/hscloud/helpers.py#L33 this can raise the same exceptions as above, which should be handled.